### DISCUSSION TITLE
Redirecting to a smaller processed image when using tiered access

### DISCUSSION THREAD

**Discussion by gverm** [cantaloupe-project/cantaloupe#565] [Q&A] [answered]
When an unauthorized user requests 'full/max' I do a redirect in 'authorize()' as instructed in the Tiered Access documentation. Is there a way to redirect it to 'full/500,'? As I currently understand it, I can only detect the redirection (using the ;1:2 in the identifier) in 'httpsource_resource_info()' and return a link to a previously saved smaller version of the requested image. Am I missing something?

**Comment by adolski** [comment, marked as answer]
Two things: 

1. `authorize()` doesn't have to redirect to a scale constraint, it can redirect to any URL. See its documentation in delegates.rb.sample. Here is a minimal, crude example:

```ruby
def authorize(context = {})
  max_allowed_width = 500
  if context['resulting_size']['width'] > max_allowed_width
    return {
      'status_code' => 302,
      'location'    => context['request_uri'].gsub('/full/max/', "/full/#{max_allowed_width},/")
    }
  end
end
```

This doesn't take e.g. rotation into account, which would affect the resulting size. And the URL path processing needs to be more robust, as it would not catch `/full/1200,`, for example. But it's the basic idea.

However,

2. The above technique is not allowed in the context of the IIIF Authentication API, if that's what you were aiming for (in mentioning tiered access). From the [API doc](https://iiif.io/api/auth/1.0/#tiered-access):

> If a server supports multiple tiers of access, then it must use a different identifier for each Description Resource and its corresponding Content Resource(s). For example, there must be different Image Information documents (/info.json) at different URIs for each tier."

**Reply by gverm** [reply]
Thank you very much for the quick detailed response, this will definitely help solve my problem. Sorry for missing the extra documentation in the sample...
